From 6c084750b72da3ffafae1cd436927d98922ef4d1 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:37:37 -0400
Subject: [ethernet] i40e: Do not enable NTUPLE feature control in MFP mode

Message-id: <1409658042-22689-97-git-send-email-sassmann@redhat.com>
Patchwork-id: 90295
O-Subject: [RHEL7.1 PATCH 096/281] i40e: Do not enable NTUPLE feature control in MFP mode
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Anjali Singhai Jain <anjali.singhai@intel.com>

Although NTUPLE feature control is disabled by default, do not
allow user to be able to change it.

Change-ID: I4c3f95fb6dbc56b1b6cc47116ea1f72de02bc99c
Signed-off-by: Anjali Singhai Jain <anjali.singhai@intel.com>
Signed-off-by: Catherine Sullivan <catherine.sullivan@intel.com>
Tested-by: Kavindya Deegala <kavindya.s.deegala@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 2e86a0b6601bfafd8497a17d1dc25ceb300aa123)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_main.c b/drivers/net/ethernet/intel/i40e/i40e_main.c
index a5ebf42..153d7e5 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_main.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_main.c
@@ -6780,10 +6780,12 @@ static int i40e_config_netdev(struct i40e_vsi *vsi)
       NETIF_F_TSO_ECN        |
       NETIF_F_TSO6         |
       NETIF_F_RXCSUM        |
-      NETIF_F_NTUPLE        |
       NETIF_F_RXHASH        |
       0;
 
+ if (!(pf->flags & I40E_FLAG_MFP_ENABLED))
+  netdev->features |= NETIF_F_NTUPLE;
+
  /* copy netdev features into list of user selectable features */
  netdev->hw_features |= netdev->features;
 
-- 
1.7.1